src/clients/python/examples/image_client.py

```diff
@@ -277,32 +204,22 @@ def postprocess(results, output_name, batch_size, batching):
 
 
 def requestGenerator(batched_image_data, input_name, output_name, dtype, FLAGS):
-
     # Set the input data
-    inputs = []
-    if FLAGS.protocol.lower() == "grpc":
-        inputs.append(
-            grpcclient.InferInput(input_name, batched_image_data.shape, dtype))
-        inputs[0].set_data_from_numpy(batched_image_data)
-    else:
-        inputs.append(
-            httpclient.InferInput(input_name, batched_image_data.shape, dtype))
-        inputs[0].set_data_from_numpy(batched_image_data, binary_data=True)
+    inputs = [grpcclient.InferInput(input_name, batched_image_data.shape, dtype)]
+    inputs[0].set_data_from_numpy(batched_image_data)
 
-    outputs = []
-    if FLAGS.protocol.lower() == "grpc":
-        outputs.append(
-            grpcclient.InferRequestedOutput(output_name,
-                                            class_count=FLAGS.classes))
-    else:
-        outputs.append(
-            httpclient.InferRequestedOutput(output_name,
-                                            binary_data=True,
-                                            class_count=FLAGS.classes))
+    outputs = [grpcclient.InferRequestedOutput(output_name, class_count=FLAGS.classes)]
```

Comment:
Have you tested this change against HTTP protocol? The input and output objects differs for the two protocols. We need to set `binary_data=true` for better performance.

Reply:
https://github.com/triton-inference-server/server/blob/master/src/clients/python/library/tritonclient/http/__init__.py#L1506
https://github.com/triton-inference-server/server/blob/master/src/clients/python/library/tritonclient/http/__init__.py#L1352

@tanmayv25 Thanks for your comment. I have totally tested this change against HTTP protocol.

If you see the line of the code above, `binary_data=True` will be filled with default value `True`

This code is an example of usage of triton inference. I think if it is more simple, people can use more easily.

Reply:
I am seeing the following error when using http protocol
```
Traceback (most recent call last):
  File "../clients/image_client.py", line 427, in <module>
    triton_client.infer(FLAGS.model_name,
  File "/usr/local/lib/python3.8/dist-packages/tritonclient/http/__init__.py", line 1074, in infer
    request_body, json_size = _get_inference_request(
  File "/usr/local/lib/python3.8/dist-packages/tritonclient/http/__init__.py", line 110, in _get_inference_request
    request_body = json.dumps(infer_request)
TypeError: name: "input"
datatype: "FP32"
shape: 1
shape: 299
shape: 299
shape: 3
 is not JSON serializable
```
This is because you are using InferInput and InferRequestedOutput objects from grpcclient module. Instead, in case of http you must use httpclient.InferInput and httpclient.InferRequestedOutput object.

Reply:
https://github.com/triton-inference-server/server/pull/2651/commits/fb0630f55af113b9e902dcf054f75cd50b86bc05